sdks/python/apache_beam/pvalue.py

```diff
@@ -288,6 +290,80 @@ def _view_options(self):
   def element_type(self):
     return typehints.Any
 
+  # TODO(robertwb): Get rid of _from_runtime_iterable and _view_options
+  # in favor of _side_input_data().
+  def _side_input_data(self):
```

Comment:
<!--new_thread; commit:3dfe862220afbc5e57d7ac674a329cb5e66164cf; resolved:0-->
Is the default implementation only for `AsSingleton`?

Reply:
It would work for any of them, but I left this here to show (and test) the relationship.